lending_pool/src/utils.rs

```diff
@@ -0,0 +1,427 @@
+
+        for bp in borrow_position_iter {
+            if bp.token_id == token_id {
+                self.borrow_position().swap_remove(&bp);
+                return bp;
+            }
+        }
+        BorrowPosition::new(
+            token_id,
+            BigUint::zero(),
+            account_position,
+            self.blockchain().get_block_round(),
+            BigUint::from(BP),
+        )
+    }
+
+    #[inline]
+    #[view(getTotalCollateralAvailableForToken)]
+    fn get_collateral_available_for_token(&self, account_position: u64, token_id: TokenIdentifier) -> BigUint {
+        let mut deposited_amount_in_dollars = BigUint::zero();
+        let deposit_positions = self.deposit_position();
+        let deposit_position_iter = deposit_positions
+            .iter()
+            .filter(|dp| dp.owner_nonce == account_position && dp.token_id == token_id);
+
+        for dp in deposit_position_iter {
+            let dp_data = self.get_token_price_data(dp.token_id);
+            deposited_amount_in_dollars += dp.amount * dp_data.price;
+        }
+
+        deposited_amount_in_dollars
+    }
+
+    #[inline]
+    #[view(getTotalCollateralAvailable)]
+    fn get_total_collateral_available(&self, account_position: u64) -> BigUint {
+        let mut deposited_amount_in_dollars = BigUint::zero();
+        let deposit_positions = self.deposit_position();
+        let deposit_position_iter = deposit_positions
+            .iter()
+            .filter(|dp| dp.owner_nonce == account_position);
+
+        for dp in deposit_position_iter {
+            let dp_data = self.get_token_price_data(dp.token_id);
+            deposited_amount_in_dollars += dp.amount * dp_data.price;
+        }
+
+        deposited_amount_in_dollars
+    }
+
+    #[view(getTotalBorrowedAmount)]
+    fn get_total_borrowed_amount(&self, account_position: u64) -> BigUint {
+        let mut total_borrowed_amount = BigUint::zero();
+        let borrow_positions = self.borrow_position();
+        let borrow_position_iter = borrow_positions
+            .iter()
+            .filter(|bp| bp.owner_nonce == account_position);
+
+        for bp in borrow_position_iter {
+            let bp_data = self.get_token_price_data(bp.token_id);
+            total_borrowed_amount += bp.amount * bp_data.price;
+        }
+
+        total_borrowed_amount
+    }
+
+    fn send_amount_in_dollars_to_liquidator(
+        &self,
+        initial_caller: ManagedAddress,
+        liquidatee_account_nonce: u64,
+        amount_to_return_to_liquidator_in_dollars: BigUint,
+    ) {
+        let mut amount_to_send = amount_to_return_to_liquidator_in_dollars;
+        let deposit_positions = self.deposit_position();
+        let deposit_position_iter = deposit_positions
+            .iter()
+            .filter(|dp| dp.owner_nonce == liquidatee_account_nonce);
+
+        // Send amount_to_return_in_dollars to initial_caller
+        for mut dp in deposit_position_iter {
```

Comment:
I'm not sure this approach is correct. Liquidation at the moment seems to be done in the arbitrary order of positions. I think it should be done in a specific order. Maybe riskier assets first? Smaller positions first? Bigger positions first? Will have to discuss which is best.

Reply:
Also, might be best to cut a slice from each position. Up for debate.

Reply:
Approached in https://github.com/ElrondNetwork/elrond-lend-rs/pull/64